front/src/modules/ui/table/hooks/useTableColumns.ts

```diff
@@ -36,5 +36,83 @@ export const useTableColumns = () => {
     [tableColumnsByKey, tableColumns, setTableColumns],
   );
 
-  return { handleColumnVisibilityChange };
+  const handleColumnMoveLeftChange = useCallback(
+    (column: ColumnDefinition<ViewFieldMetadata>) => {
+      const tableColumnIndex = tableColumns.findIndex(
+        (tableColumn) => tableColumn.key === column.key,
+      );
+      if (tableColumnIndex >= 0) {
+        const lastColumn = tableColumns[tableColumnIndex - 1];
+        const nextToLastColumn = tableColumns[tableColumnIndex - 2];
+        if (nextToLastColumn) {
+          const newIndex = (lastColumn.index + nextToLastColumn.index) / 2;
+          const updatedColumns = tableColumns
+            .map((previousColumn) =>
+              previousColumn.key === column.key
+                ? { ...previousColumn, index: newIndex }
+                : previousColumn,
+            )
```

Comment:
We want our desired column to be placed between the previous two columns. Here the `lastColumn` and `nextToLastColumn` variables holds those indexes. We calculate the mean between those indexes and assign the calculated value to our current column which we want to move and update the state.